src/api/app/controllers/webui/staging_workflows_controller.rb

```diff
@@ -40,6 +39,26 @@ def show
     @project = @staging_workflow.project
   end
 
+  def destroy
+    @staging_workflow = StagingWorkflow.find_by(id: params[:id])
+    @project = @staging_workflow.project
+
+    #TODO: authorization
+    
+    if @staging_workflow.staging_projects
+      @staging_workflow.staged_requests.update_all staging_project_id: nil
+      @staging_workflow.staging_projects.where(id: params[:staging_project_ids]).destroy_all 
```

Comment:
This can take a long time - imagine 10 staging projects with 10 packages each, mean like O(100) backend requests. Some feedback to the user what is going on is basically required. Otherwise you stare at that ajax dialog and wondering if you hit the right button - and reload the page and create these fancy deadlock errbit exceptions.

Reply:
@coolo Makes sense, a spinner should help

Reply:
done :dancer: